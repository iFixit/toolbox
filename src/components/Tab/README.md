### Examples

#### Default Tab
```
<Tab>All</Tab>
```

#### Active Tab
```
<Tab active>All</Tab>
```

#### Element prop

The `Tab` component accepts an `is` prop to change the underlying HTML element on a per-instance basis. This is useful for ensuring semantic markup, while keeping styles decoupled.

```
<Tab is="span">All</Tab>
```