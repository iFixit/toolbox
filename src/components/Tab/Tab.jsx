import React from 'react';
import { bool, element, func, string, oneOfType } from 'prop-types';
import glamorous from 'glamorous';

import constants from '../../constants';

const { spacing, lineHeight, color } = constants;

const TabContainer = glamorous.a(
   {
      display: 'inline-block',
      boxSizing: 'border-box',
      padding: `${spacing[3]} 0`,
      marginRight: spacing[4],
      fontWeight: 700,
      lineHeight: lineHeight.title,
      textDecoration: 'none',
      color: color.grayAlpha[9],

      '&:hover': {
         color: color.blue[4],
      },
   },
   ({ active }) => active && {
      color: color.blue[4],
      borderBottom: `3px solid ${color.blue[4]}`,
   },
);

const Tab = ({ is, ...props }) => {
   const Component = TabContainer.withComponent(is);
   return <Component {...props} />;
};

Tab.propTypes = {
   active: bool,
   /** HTML element string or React component to render */
   is: oneOfType([string, func, element]),
};

Tab.defaultProps = {
   active: false,
   is: 'a',
};

export default Tab;
