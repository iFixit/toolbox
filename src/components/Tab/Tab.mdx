---
name: Tab
route: /components/tab
menu: Components
---

import Tab from './Tab'
import { Playground, PropsTable } from 'docz'

# Tab

## Examples

### Default Tab
<Playground>
  <Tab>All</Tab>
</Playground>

### Active Tab
<Playground>
  <Tab active>All</Tab>
</Playground>

### Element prop

The `Tab` component accepts an `is` prop to change the underlying HTML element on a per-instance basis. This is useful for ensuring semantic markup, while keeping styles decoupled.

<Playground>
  <Tab is="span">All</Tab>
</Playground>

## Props

<PropsTable of={Tab} />
