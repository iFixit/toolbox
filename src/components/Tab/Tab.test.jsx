import React from 'react';
import renderer from 'react-test-renderer';
import Tab from './Tab';

test('renders without crashing', () => {
   const tree = renderer.create(<Tab>Test</Tab>).toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders when active', () => {
   const tree = renderer.create(<Tab active>Test</Tab>).toJSON();
   expect(tree).toMatchSnapshot();
});
