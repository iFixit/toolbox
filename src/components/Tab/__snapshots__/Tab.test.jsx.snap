// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when active 1`] = `
.glamor-0,
[data-glamor-0] {
  flex: 0 0 auto;
  display: inline-block;
  box-sizing: border-box;
  padding: 12px;
  margin-right: 12px;
  font-size: 16px;
  font-weight: 700;
  line-height: 1.25;
  text-decoration: none;
  color: #0071CE;
  border-top: 3px solid transparent;
  border-bottom: 3px solid transparent;
  transition: all 150ms ease-in-out;
  outline: none;
  border-bottom-color: #0071CE;
  -webkit-flex: 0 0 auto;
  -webkit-transition: all 150ms ease-in-out;
  -moz-transition: all 150ms ease-in-out;
}

.glamor-0:hover,
[data-glamor-0]:hover {
  color: #0071CE;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  border-bottom-color: rgba(0, 3, 6, 0.26);
}

<a
  className="glamor-0"
>
  Test
</a>
`;

exports[`renders without crashing 1`] = `
.glamor-0,
[data-glamor-0] {
  flex: 0 0 auto;
  display: inline-block;
  box-sizing: border-box;
  padding: 12px;
  margin-right: 12px;
  font-size: 16px;
  font-weight: 700;
  line-height: 1.25;
  text-decoration: none;
  color: rgba(0, 3, 6, 0.87);
  border-top: 3px solid transparent;
  border-bottom: 3px solid transparent;
  transition: all 150ms ease-in-out;
  outline: none;
  -webkit-flex: 0 0 auto;
  -webkit-transition: all 150ms ease-in-out;
  -moz-transition: all 150ms ease-in-out;
}

.glamor-0:hover,
[data-glamor-0]:hover {
  color: #0071CE;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  border-bottom-color: rgba(0, 3, 6, 0.26);
}

<a
  className="glamor-0"
>
  Test
</a>
`;
